### ISSUE TITLE
How to use clang++ instead clang to compile SVF 

### ISSUE THREAD

**Issue by nadal11919** [SVF-tools/SVF#648]
when I run the command“source ./build.sh debug“ to build SVF，the errors occured like this：
  
"vtable for llvm::Pass", referenced from:
      llvm::Pass::Pass(llvm::PassKind, char&) in libSvf.a(SymbolTableInfo.cpp.o)
      llvm::Pass::Pass(llvm::PassKind, char&) in libSvf.a(BreakConstantExpr.cpp.o)
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/saber] Error 1
make[1]: *** [tools/SABER/CMakeFiles/saber.dir/all] Error 2
make: *** [all] Error 2

Someone online tell me that I should use clang++ instead clang to build this project，but I do not know how to set it？ Or are there any other methods to fix it？

**Comment by yuleisui** [comment]
Not sure about your errors. You can take a good at github workflow of the machine config: https://github.com/SVF-tools/SVF/blob/51bc1a0b2193864ae52fbe50da369b6adc7aa246/.github/workflows/github-action.yml#L36

You can also use SVF as a lib which might be easier to get started https://github.com/SVF-tools/SVF-example

**Comment by nadal11919** [comment]
thanks for your reply！But I forgot to point out my environment is macOS11.4.So I really do not know how to operate the command"sudo apt-get ..."  of the first step in the setup guide page for setting up  SVF in VScode.I gave up to build it in macOS and tried to  use Linux to build the SVF.

Sorry I am a new learner and I have other questions for the CallGraph: 

I am confused about the difference about the direct calls/ indirect calls/ virtual calls in SVF. Do the indirect calls mean the use of function pointers and the virtual calls mean the use of class inheritance？  And I found the functions “resolveCPPIndCalls”  and "resolveIndCalls". Is the direct calls will generate the call edges when initializing？

thank you so much！
